engine/src/it/java/org/pentaho/di/pan/PanIT.java

```diff
@@ -0,0 +1,535 @@
+
+      // get log file contents
+      String logFileContent = getFileContentAsString( logFileRelativePath );
+
+      // use FINISHED_PROCESSING_ERROR_COUNT_REGEX to get execution error count
+      int errorCount = parseErrorCount( logFileContent );
+
+      assertTrue( !logFileContent.contains( FAILED_TO_INITIALIZE_ERROR_PATTERN ) &&  errorCount == 0 );
+
+      Result result = Pan.getCommandExecutor().getResult();
+      assertNotNull( result );
+      assertEquals( CommandExecutorCodes.Pan.SUCCESS.getCode(), result.getExitStatus() );
+
+    } finally {
+      // sanitize
+      File f = new File( logFileFullPath );
+      if ( f != null && f.exists() ) {
+        f.deleteOnExit();
+      }
+    }
+  }
+
+  /**
+   * Tests that step 'Transformation Executor' successfully passes parameters to the transformation.
+   * @throws Exception
+   */
+  @Test
+  public void testFileTransExecutorSubParameters() throws Exception {
+
+    /* NOTE:
+     * '03_Trans_to_TransExec.ktr' should call 'transformation_value.ktr'. The log of '' should contain the snippet
+     * Write to log.0 - DATE_ID = 2018-04-01, where DATE_ID is passed as a parameter from step 'Transformation Executor'
+     * with a value specified from step 'Generate Rows'.
+     */
+
+    long now = System.currentTimeMillis();
+
+    String testKTRRelativePath = "test_parameters/executor/03_Trans_to_TransExec.ktr";
+    String testKTRFullPath = getRelativePathKTR( testKTRRelativePath );
+    String logFileRelativePath = testKTRRelativePath + "." + now + ".log";
+    String logFileFullPath = testKTRFullPath + "." + now + ".log";
+
+    try {
+
+      Pan.main( new String[] { "/file:" + testKTRFullPath, "/level:Basic", "/logfile:" + logFileFullPath } );
+
+    } catch ( SecurityException e ) {
+      // All OK / expected: SecurityException is purposely thrown when Pan triggers System.exitJVM()
+
+      // get log file contents
+      String logFileContent = getFileContentAsString( logFileRelativePath );
+
+      // use FINISHED_PROCESSING_ERROR_COUNT_REGEX to get execution error count
+      int errorCount = parseErrorCount( logFileContent );
+
+      assertTrue( !logFileContent.contains( FAILED_TO_INITIALIZE_ERROR_PATTERN ) &&  errorCount == 0 );
+      assertTrue( logFileContent.contains( "DATE_ID = 2018-04-01" ) ); // most significant assert
+
+      Result result = Pan.getCommandExecutor().getResult();
+      assertNotNull( result );
+      assertEquals( CommandExecutorCodes.Pan.SUCCESS.getCode(), result.getExitStatus() );
+
+    } finally {
+      // sanitize
+      File f = new File( logFileFullPath );
+      if ( f != null && f.exists() ) {
+        f.deleteOnExit();
+      }
+    }
+  }
+
+  /**
+   * Get file contentsf from absolute path.
+   * @param relativePath
+   * @return
+   */
+  private String getFileContentAsString( String relativePath ) {
+
+    String content = "";
+
+    try {
+      BufferedReader rd = new BufferedReader(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *rd*. Execution paths that do not contain closure statements were detected. Ensure *rd* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).